PubSub/src/Subscription.php

```diff
@@ -690,18 +702,35 @@ public function pull(array $options = [])
      *     $subscription->acknowledge($message);
      * }
      * ```
+     * ```
+     * $messages = $subscription->pull();
+     *
+     * foreach ($messages as $message) {
+     *     $failedMsgs = $subscription->acknowledge($message, ['returnFailures' => true]);
+     *
+     *     // Either log or store the $failedMsgs to be retried later
+     * }
+     * ```
      *
      * @codingStandardsIgnoreStart
      * @see https://cloud.google.com/pubsub/docs/reference/rest/v1/projects.subscriptions/acknowledge Acknowledge Message
      * @codingStandardsIgnoreEnd
      *
      * @param Message $message A message object.
-     * @param array $options [optional] Configuration Options
-     * @return void
+     * @param array $options [optional] {
+     *      Configuration Options
+     *
+     *      @type bool $returnFailures If set, and if a message is failed with a
+     *            temporary failure code, it will be retried with an exponential delay. This will also make sure
+     *            that the permanently failed message is returned to the caller. This is only true for a
+     *            subscription with 'Exactly Once Delivery' enabled.
+     *            Read more about EOD: https://cloud.google.com/pubsub/docs/exactly-once-delivery
+     * }
+     * @return void|array
      */
     public function acknowledge(Message $message, array $options = [])
     {
-        $this->acknowledgeBatch([$message], $options);
+        return $this->acknowledgeBatch([$message], $options);
```

Comment:
Sorry might be a dump question. Does adding "return" impact/change the behavior for existing users? It should be a no-op.

Reply:
It remains a no-op for the existing users. We are only making sure that whatever `acknowledgeBatch` returns is cascaded to the `acknowledge` returns as this method is piggy backing on it.